dts/bindings/riscv/nordic,vpr-coprocessor.yaml

```diff
@@ -18,15 +18,13 @@ properties:
     description: |
       Processor ID of the VPR core.
 
-  loader-img-dst:
+  execution-memory:
     type: phandle
     required: true
     description: |
-      Destination memory handle. Points to the area from which the VPR core
-      will start execution.
+      Memory area from which the VPR core will execute.
 
-  loader-img-src:
+  source-partition:
```

Comment:
Do we only want to support partitions, or normal memory nodes as well?

Reply:
If that's the case, I'd probably have two properties, or maybe some macrobatics in the driver (if possible?)

Reply:
Macrobatics shouldn't be too bad. I had this prepared:

```c
#define VPR_SRC_ADDR(node_id)                                                                      \
        (DT_REG_ADDR(node_id) +                                                                    \
         COND_CODE_0(DT_FIXED_PARTITION_EXISTS(node_id), (0), (DT_REG_ADDR(DT_GPARENT(node_id)))))
```

Reply:
ok, macrobatics are not that bad, will use that snippet, thanks!